packages/backend-deployer/src/cdk_error_mapper.ts

```diff
@@ -247,6 +247,17 @@ export class CdkErrorMapper {
       errorName: 'SecretNotSetError',
       classification: 'ERROR',
     },
+    {
+      // This error originates from AppSync when there are syntax/type errors or unsupported features being used in resolvers
+      errorRegex:
+        /(Resource handler returned message|BadRequestException): The code contains one or more errors./,
+      humanReadableErrorMessage:
+        'There are one or more errors with data handlers.',
+      resolutionMessage:
+        'Check your handlers defined in your data backend for syntax errors, type errors, and unsupported features of AppSync resolvers.',
+      errorName: 'AppSyncHandlerError',
+      classification: 'ERROR',
```

Comment:
There's an edge case scenario for this classification.

If data (or any of our constructs) start deploying JS resolvers (and it seems it's coming) then this might be fault not error depending on who creates the resource.